data/test/index.js

```diff
@@ -233,6 +267,71 @@ describe( 'subscribe', () => {
 
 		expect( incrementedValue ).toBe( 3 );
 	} );
+
+	it( 'subscribes to specific reducer', () => {
+		registerStore( 'one', {
+			reducer: ( state = 0 ) => state + 1,
+			actions: { increment: () => ( { type: 'increment' } ) },
+		} );
+		registerStore( 'two', {
+			reducer: ( state = 0 ) => state + 1,
+			actions: { increment: () => ( { type: 'increment' } ) },
+		} );
+
+		const oneCallback = jest.fn();
+		const twoCallback = jest.fn();
+
+		const oneUnsubscribe = subscribe( 'one', oneCallback );
+		const twoUnsubscribe = subscribe( 'two', twoCallback );
+
+		dispatch( 'one' ).increment();
+
+		expect( oneCallback ).toHaveBeenCalled();
+		expect( twoCallback ).not.toHaveBeenCalled();
+
+		oneUnsubscribe();
```

Comment:
Minor: when `expect` fails unsubscribe methods won't be executed. It shouldn't have impact on other tests though. One way to solve it is to put all unsubscribe methods in array and iterate over it using `afterEach`.

````js
let unsubscribes = [];
afterEach( () => {
    unsubscribes.forEach( ( unsubscribe ) => unsubscribe() );
} );
```

Reply:
Yeah, that's true. At the same time though, if a test fails, all bets are off. The only issue is if it impacted future failures that caused needless debugging. I'll see about adding lifecycle.

Reply:
Exactly, it might fail not related tests in the future. It’s a minor thing and really hard to implement in a clean way 🙁

Reply:
Since 84c64c8 was removed in rebase, this is no longer relevant. That said, before dropping, I did find it was pretty simple to implement a helper `subscribe` wrapper:

```js
const unsubscribes = [];
afterEach( () => {
	let unsubscribe;
	while ( ( unsubscribe = unsubscribes.shift() ) ) {
		unsubscribe();
	}
} );

function subscribeWithUnsubscribe( ...args ) {
	unsubscribes.push( subscribe( ...args ) );
}
```